src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -15,10 +15,47 @@
 namespace System
 {
     /// <summary>
-    /// Extension methods and non-generic helpers for Span and ReadOnlySpan
+    /// Extension methods and non-generic helpers for Span, ReadOnlySpan, Memory, and ReadOnlyMemory.
     /// </summary>
     public static class Span
     {
+        /// <summary>Creates a new <see cref="ReadOnlyMemory{T}"/> over the portion of the target string.</summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is a null reference (Nothing in Visual Basic).</exception>
+        public static ReadOnlyMemory<char> AsReadOnlyMemory(this string text)
```

Comment:
We don't need these extension methods here if the implementation is the exact same as what is in corefx.

Reply:
The implementation in corefx wouldn't have access to the internals these are using, i.e. the internal ctor and the internal GetObjectStartLength method.